lora/adapter_test.go

```diff
@@ -48,19 +57,32 @@ func TestPublish(t *testing.T) {
 
 	err = svc.ConnectThing(context.Background(), chanID, thingID)
 	require.Nil(t, err, fmt.Sprintf("unexpected error: %s\n", err))
+	repoCall.Unset()
+	repoCall1.Unset()
+	repoCall2.Unset()
+	repoCall3.Unset()
+	repoCall4.Unset()
 
+	repoCall = channelsRM.On("Save", context.Background(), chanID2, appID2).Return(nil)
+	repoCall1 = thingsRM.On("Save", context.Background(), thingID2, devEUI2).Return(nil)
 	err = svc.CreateChannel(context.Background(), chanID2, appID2)
 	require.Nil(t, err, fmt.Sprintf("unexpected error: %s\n", err))
 
 	err = svc.CreateThing(context.Background(), thingID2, devEUI2)
 	require.Nil(t, err, fmt.Sprintf("unexpected error: %s\n", err))
+	repoCall.Unset()
+	repoCall1.Unset()
 
 	msgBase64 := base64.StdEncoding.EncodeToString([]byte(msg))
 
 	cases := []struct {
-		desc string
-		err  error
-		msg  lora.Message
+		desc           string
+		err            error
+		msg            lora.Message
+		getThingErr    error
+		getChannelErr  error
+		connectionsErr error
+		publishErr     error
```

Comment:
These errors are mostly not used (i.e. only `publishErr` is checked). All the errors should be asserted and we need cases for different repo call errors.